### PR TITLE
fix(trigger): Fix bug with trying to read trigger during initialization

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy [hunk 1]
```diff
@@ -61,11 +61,19 @@ class DependentPipelineStarter implements ApplicationContextAware {
     pipelineConfig.trigger = new PipelineTrigger(
       parentPipeline,
       parentPipelineStageId,
-      principal?.username ?: user ?: "[anonymous]",
-      buildParameters(pipelineConfig, suppliedParameters),
```
Comment (robfletcher) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy]: This was the problem. It was passing `pipelineConfig` to this method which will read `trigger` from it, but it's doing so while assigning `trigger`. Never going to work.

**Comment by ajordens** [comment]
LGTM ... this is basically a restoration of the order in which we were building `trigger.parameters` right?

**Comment by robfletcher** [comment]
Right.

File: orca-front50/src/test/groovy/com/netflix/spinnaker/orca/front50/pipeline/EnabledPipelineValidatorSpec.groovy [hunk 2]
```diff
@@ -121,7 +121,7 @@ class EnabledPipelineValidatorSpec extends Specification {
     execution = pipeline {
       application = "whatever"
       pipelineConfigId = "1337"
-      trigger = new PipelineTrigger(null, [strategy: true])
+      trigger = new PipelineTrigger(pipeline {}, [strategy: true])
```
Comment (robfletcher) [inline orca-front50/src/test/groovy/com/netflix/spinnaker/orca/front50/pipeline/EnabledPipelineValidatorSpec.groovy]: Running these tests in IntelliJ could fail if compiler `@Nonnull` checks are enforced.

**Comment by ezimanyi** [comment]
LGTM

**Pull request merged**